src/bitcask.erl

```diff
@@ -1181,17 +1189,35 @@ scan_key_files([Filename | Rest], KeyDir, Acc, CloseFile, KT) ->
             %% tombstones or data errors.  Otherwise we risk of
             %% reusing the file id for new data.
             _ = bitcask_nifs:increment_file_id(KeyDir, FileTstamp),
-            F = fun({tombstone, K}, _Tstamp, {_Offset, _TotalSz}, _) ->
-                        _ = bitcask_nifs:keydir_remove(KeyDir, KT(K));
-                   (K, Tstamp, {Offset, TotalSz}, _) ->
-                        bitcask_nifs:keydir_put(KeyDir,
-                                                KT(K),
-                                                FileTstamp,
-                                                TotalSz,
-                                                Offset,
-                                                Tstamp,
-                                                bitcask_time:tstamp(),
-                                                false)
+            F = fun({tombstone, K0}, _Tstamp, {_Offset, _TotalSz}, _) ->
+                        K = try KT(K0) catch TxErr -> {key_tx_error, TxErr} end,
+                        case K of
+                            {key_tx_error, KeyTxErr} ->
+                                error_logger:error_msg("Invalid key on load ~p: ~p",
+                                                       [K0, KeyTxErr]),
+                                ok;
+                            _ ->
+                                bitcask_nifs:keydir_remove(KeyDir, KT(K))
+                        end,
+                        ok;
+                   (K0, Tstamp, {Offset, TotalSz}, _) ->
+                        K = try KT(K0) catch TxErr -> {key_tx_error, TxErr} end,
+                        case K of
+                            {key_tx_error, KeyTxErr} ->
+                                error_logger:error_msg("Invalid key on load ~p: ~p",
+                                                       [K0, KeyTxErr]),
+                                ok;
```

Comment:
Let's have consistency here, please. We either close clauses with `ok` or we do it as the final return value after the case statement. I'm fine with either but they should be the same.